tools/get_package_version.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+# Copyright 2019 The Project Buendia Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License.  You may obtain a copy
+# of the License at: http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distrib-
+# uted under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, either express or implied.  See the License for
+# specific language governing permissions and limitations under the License.
+
+# get_package_version.sh tries to come up with a sensible package version
+# scheme for the current build situation. The logic we're following is that
+# only release packages built in CI get a formal version number; all other
+# builds should be marked as such.
+#
+# Also, we should treat CI builds as more recent than release builds for the
+# same version tag; however, we should NOT treat manual builds as more recent
+# than either. The logic there is that we do want to allow deployment of
+# replicable pre-release builds, but we don't want accidental deployment of 
+# developer builds. The most recent version is retained for the sanity of the
+# developer.
```

Comment:
Somewhere (maybe here, maybe on the wiki or a README.md) there should be documentation of the numbering format that's optimized for readers who see these numbers but don't care how they are generated, e.g.:

> There are three kinds of package version numbers:
>     0.8.4 — This is a public release.
>     0.8.4+b15 — This is CircleCI build number 15, and the most recent preceding release is 0.8.4.
>     0.8.4~g123456 — This is a release built at Git commit number 123456, and the most recent preceding release is 0.8.4.

>

Reply:
I'll take care of this separately

Reply:
https://github.com/projectbuendia/buendia/wiki/Software-Package-Management#project-version-numbering-conventions